@cawfree/local pods

cawfree opened the pull request:
- Added `Rainbow-Internals` Pod which can be used to quickly playground external dependencies.
    - These files are ignored by `git` even when changed. (They shouldn't be persisted)
  - Added `Extras.json` placeholder for custom runtime attributes.
    - Again, these shouldn't be persisted!
  -   Added `RAINBOW_SCRIPTS_APP_IOS_PREBUILD_HOOK` to notify CI of build invocation

brunobar79 reviewed (commented):
This is great. Thanks @cawfree  🙏 

Couple of comments:

- While I do appreciate the intention of keeping js & native deps in sync, I think the team doesn't prefer to run` pod install` under `yarn.` => `postinstall.sh`
The majority of the time we're not updating native dependencies so this is adding a considerable delay to just updating JS deps.

Can you rever that part of the PR?

- I don't think the Internals pods use case justify its existance. Unless I'm missing something I think we can live with it just fine.

cawfree commented:
> This is great. Thanks @cawfree  🙏 
> 
> 
> 
> Couple of comments:
> 
> 
> 
> - While I do appreciate the intention of keeping js & native deps in sync, I think the team doesn't prefer to run` pod install` under `yarn.` => `postinstall.sh`
> 
> The majority of the time we're not updating native dependencies so this is adding a considerable delay to just updating JS deps.
> 
> 
> 
> Can you rever that part of the PR?
> 
> 

No worries! I'll look into an alternative. 👀

cawfree commented:
- Removed `yarn install-bundle && yarn install-pods` from `postinstall.sh`.
  - Executed the iOS prebuild hook conditionally; this now only occurs on MacOS build machines.
  - Removed trailing semicolon from `pre-commit.sh`.

brunobar79 commented on README.md:
```diff
@@ -45,6 +45,7 @@ If you are new to React Native, this is a helpful introduction: https://facebook
 
 9. Run the project by clicking the play button.
 
+
```
Can drop this change

brunobar79 commented on ios/Internals/.gitignore:
```diff
@@ -0,0 +1,60 @@
+# Expo
+.expo/*
+
+# generated by bob
```
Wut?

brunobar79 commented on ios/Internals/.gitignore:
```diff
@@ -0,0 +1,60 @@
+*.xcuserstate
+project.xcworkspace
+
+# Android/IJ
```
Can remove this

brunobar79 commented on ios/Internals/.gitignore:
```diff
@@ -0,0 +1,60 @@
+
+# Cocoapods
+#
+example/ios/Pods
```
And this?

brunobar79 commented on ios/Internals/.gitignore:
```diff
@@ -0,0 +1,60 @@
+yarn-error.log
+
+# BUCK
+buck-out/
```
Probably too

brunobar79 commented on ios/Internals/.gitignore:
```diff
@@ -0,0 +1,60 @@
+android/keystores/debug.keystore
+
+# Expo
+.expo/*
```
Too

brunobar79 commented on package.json:
```diff
@@ -33,7 +33,7 @@
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+      "pre-commit": "./scripts/pre-commit.sh"
```
What's the reason behind this?

cawfree replied:
Originally I was performing some configuration actions on `pre-commit` which required a dedicated file, but eventually these were moved to `postinstall.sh`. I thought there might be a chance it would be useful to hang onto a dedicated file, but I'll revert this too!

brunobar79 commented on scripts/postinstall.sh:
```diff
@@ -1,7 +1,30 @@
+esac
+
+# MacOS
+if [[ "$MACHINE" == Mac ]]; then
```
We build android from iOS too locally. Don't think this is gonna work

brunobar79 reviewed (commented):
Few more comments

brunobar79 reviewed (approved):
🚀
